web/packages/teleport/src/SessionRecordings/view/Timeline/renderers/ResizeEventsRenderer.ts

```diff
@@ -0,0 +1,528 @@
+
+      if (otherEvent.textMetrics.width > 0) {
+        const maxTextOffset = Math.max(
+          textPadding,
+          width - otherEvent.textMetrics.width - textPadding
+        );
+        textOffset = Math.min(defaultTextOffset, maxTextOffset);
+      }
+
+      const otherX = otherEvent.startPosition + textOffset;
+      const otherTextWidth = otherEvent.textMetrics.width;
+
+      const otherRectX = otherX - TEXT_PADDING_X;
+      const otherRectWidth = otherTextWidth + TEXT_PADDING_X * 2;
+
+      const otherTextHeight =
+        otherEvent.textMetrics.actualBoundingBoxAscent +
+        otherEvent.textMetrics.actualBoundingBoxDescent;
+      const otherRectY =
+        otherY -
+        otherEvent.textMetrics.actualBoundingBoxAscent -
+        TEXT_PADDING_Y;
+      const otherRectHeight = otherTextHeight + TEXT_PADDING_Y * 2;
+
+      // Check if the vertical line intersects with the other event's bounding box
+      if (
+        lineX >= otherRectX &&
+        lineX <= otherRectX + otherRectWidth &&
+        lineStartY <= otherRectY + otherRectHeight &&
+        lineEndY >= otherRectY
+      ) {
+        const newSegments: LineSegment[] = [];
+
+        for (const segment of segments) {
+          // If the segment intersects with the other event's bounding box, split it
+          if (segment.start < otherRectY && segment.end > otherRectY) {
+            newSegments.push({ end: otherRectY, start: segment.start });
+          }
+
+          // If the segment intersects with the bottom of the other event's bounding box, split it
+          if (
+            segment.start < otherRectY + otherRectHeight &&
+            segment.end > otherRectY + otherRectHeight
+          ) {
+            newSegments.push({
+              end: segment.end,
+              start: otherRectY + otherRectHeight,
+            });
+          }
+
+          // If the segment is completely above or below the other event's bounding box, keep it as is
+          if (
+            segment.start >= otherRectY + otherRectHeight ||
+            segment.end <= otherRectY
+          ) {
+            newSegments.push(segment);
+          }
+        }
+
+        segments = newSegments;
+      }
+    }
+
+    return segments;
+  }
+
+  // getEventPositions filters and positions events based on the current view offset and container size.
+  // It returns events that are within the visible area plus a buffer, and assigns them to rows to avoid overlap.
+  private getEventPositions(
+    offset: number,
+    containerWidth: number,
+    containerHeight: number
+  ): EventWithCalculatedPosition[] {
+    const viewStart = -offset;
+    const viewEnd = -offset + containerWidth;
+    const buffer = containerWidth / 2;
+
+    const activeEvents = this.allEvents.filter(
+      event =>
+        event.endPosition > viewStart - buffer &&
+        event.startPosition < viewEnd + buffer
+    );
+
+    const sortedEvents = [...activeEvents].toSorted(
```

Comment:
Nit: if using `toSorted` (as opposed to `sort`), you don't have to clone the `activeEvents` array, as `toSorted` doesn't sort in place.